sympy/solvers/ode/tests/test_ode.py

```diff
@@ -889,9 +889,11 @@ def test_2nd_power_series_ordinary():
     assert checkodesol(eq, sol) == (True, 0)
 
     eq = (1 + x**2)*(f(x).diff(x, 2)) + 2*x*(f(x).diff(x)) -2*f(x)
-    assert classify_ode(eq) == ('2nd_power_series_ordinary',)
+    assert classify_ode(eq) == ('2nd_hypergeometric', '2nd_hypergeometric_Integral',
+    '2nd_power_series_ordinary')
+
     sol = Eq(f(x), C2*(-x**4/3 + x**2 + 1) + C1*x + O(x**6))
```

Comment:
@oscarbenjamin I am not sure about this example whether it should match this equation with `2nd_hypergeometric` hint. Although it returns a solution when called with dsolve.
```python
>>> eq = (1 + x**2)*(f(x).diff(x, 2)) + 2*x*(f(x).diff(x)) -2*f(x)
>>> dsolve(eq)
Eq(f(x), sqrt(x**2 + 1)*(C1*sqrt(x**2/(x**2 + 1))*hyper((0, 0), (3/2,), x**2/(x**2 + 1)) + C2*hyper((-1/2, -1/2), (1/2,), x**2/(x**2 + 1)))*(x**2)**(1/4)/sqrt(x))
```
But when I call checkodesol(eq, sol), It hangs.

According to earlier matching code, it has conditions such that whener a example has `2nd_power_series_ordinary` as a hint, then it will not check for `2nd_hypergeometric`

Reply:
We basically never want the series solution if one of the other solvers can give a non-series solution so the earlier matching code was incorrect.